39 13 release blog (#340)
flowforge/website: src/blog/2023/01/flowforge-1-3-0-released.md

---
title: FlowForge 1.3 is now available, share your flows through our new team libraries and much more
subtitle: Our first release of 2023 with some great new features to try out, happy new year from everyone at FlowForge!

> description: 

Reviewer: ![Screenshot 2023-01-19 at 13 33 11](https://user-images.githubusercontent.com/2529595/213443936-71a3221d-ded1-4fbb-975f-36bbd64f0645.png)

The `null` is shown if this field is empty, can we add something?

Author: Fixed in the commit I'm just putting together.

date: 2023-01-19 18:00:00.0
authors: ["rob-marcer"]
video: ey3xv5j5x7k
---

Share your flows via team libraries, control access to your Node-RED dashboards using FlowForge credentials, and filter your audit logs by users and actions.

<!--more-->

We're pleased to announce version 1.3 is now available! Due to the recent holiday season, most of our team have been away from their desks but we still have some great new features to share. Keep reading for the details of what's in this release or you can watch our 1 minute roundup video of the new release above.

To make it easy for everyone to experience FlowForge, we are introducing a new [free 30-day trial](https://app.flowforge.com/account/create). With this trial, you can experience the power of using FlowForge to quickly deliver Node-RED applications in a reliable, repeatable, collaborative, and secure manner. To get your trial simply [sign up for a new FlowForge team](https://app.flowforge.com/account/create).

## Features

> [Share your flows via team libraries](https://github.com/flowforge/flowforge/issues/237)

Reviewer 2: This should be a title so it is on its own line.

Author: Fixed for all three main stories in the commit I'm working on now

FlowForge has now added the ability for you to share your flows via the import and export features in Node-RED. Once you export a flow everyone else in your FlowForge team will be able to import your work into their projects. You can see a demonstration of this new feature in [the video](https://youtu.be/B7XK3TUklUU) below.

<div><iframe width="560" height="315" src="https://www.youtube.com/embed/B7XK3TUklUU" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

> [Control access to your Node-RED dashboards using FlowForge credentials](https://github.com/flowforge/flowforge/issues/1325)

Reviewer 2: Similar to above, should be a title

Author: Fixed

In FlowForge 0.10 we added the ability to secure endpoints created within your FlowForge projects. This allows you to create dashboards or APIs and limit who can access them. In 1.3 we've added the ability for you to limit access to those same resources based on the visitor having a user account on your FlowForge team. You can see a demonstration of this new feature in [the video](https://youtu.be/JRk-Cf7eNIo) below.

<div><iframe width="560" height="315" src="https://www.youtube.com/embed/JRk-Cf7eNIo" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

> [Filter your audit logs for easier reading](https://github.com/flowforge/flowforge/issues/1448)

Reviewer 2: Needs to be a title

Author: Fixed

In FlowForge 1.3 we’ve added the ability to filter your admin logs by user or action type. We think this is a great new feature which will help admins have confidence that they will be able to review the audit logs quickly when needed. You can see a demonstration of this new feature in [the video](https://youtu.be/p0Vuy5x42Go) below.

<div><iframe width="560" height="315" src="https://www.youtube.com/embed/p0Vuy5x42Go" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

## Improvements

> [Allow installation of FlowForge on devices which can't access npm](https://github.com/flowforge/flowforge-device-agent/issues/45)

Reviewer 2: Need to be a title

We've had feedback from customers that in some cases they want to use FlowForge devices on hardware which cannot access [Node Package Manager](https://www.npmjs.com/) (npm). In a standard configuration of Node-RED, access to npm is mandatory to run your flows. In FlowForge 1.3.0 we've added the ability for you to import all the data usually installed from npm without your devices having access to the npm service.

## Bug Fixes

> We've fixed the following bugs in this release.
> 
> - Project status UI sometimes getting stuck when restarting [#1232](https://github.com/flowforge/flowforge/issues/1232)
> 
> - SSO users asked to click link in email to verify [#1543](https://github.com/flowforge/flowforge/issues/1543)
> 
> - SSO users unable to edit settings [#1542](https://github.com/flowforge/flowforge/issues/1542)
> 
> - SSO users not redirected to editor when signing in [#1481](https://github.com/flowforge/flowforge/issues/1481)

Reviewer: Maybe a better style?

```suggestion
A couple of bugs were fixed in 1.3 too; one bug of those got the project status UI [stuck when restarting](https://github.com/flowforge/flowforge/issues/1232). A few around Single Sign in were fixed too; firstly one asking users to verify their email address, which isn't required with SSO. Another disallowed SSO users to their profile details, an unintended consequence from [disallowing these users to update their email address](https://github.com/flowforge/flowforge/issues/1542). Lastly there was a broken redirect [when signing in](https://github.com/flowforge/flowforge/issues/1481).
```

Author: Agreed, adding it in as part of my commit.

## Contributors

We'd like to thank the following for their contributions to this release:

[flecoufle](https://github.com/flecoufle) for their work on [#89](https://github.com/flowforge/helm/pull/89)

As an open-source project, we welcome community involvement in what we're building.
If you're interested in contributing, checkout our [guide in the docs](https://flowforge.com/docs/contribute/).

### Try it out

We're confident you can have self managed FlowForge running locally in under 30 minutes.
You can install our [local build](https://flowforge.com/docs/install/local/), use [Docker](https://flowforge.com/docs/install/docker/), or [Kubernetes](https://flowforge.com/docs/install/kubernetes/).

> If you'd rather use our hosted offering: [Sign up for FlowForge Cloud](https://app.flowforge.com/account/create) to get your first project free for a month.

Reviewer: ```suggestion
If you'd rather use our hosted offering: [Get started for free](https://app.flowforge.com/account/create) on FlowForge Cloud.
```

### Upgrading FlowForge

[FlowForge Cloud](https://app.flowforge.com) is already running 1.3.

If you installed a previous version of FlowForge and want to upgrade, our documentation provides a
guide for [upgrading your FlowForge instance](https://flowforge.com/docs/upgrade/).

### Getting help

> Please check FlowForge's [documentation](https://flowforge.com/docs/) as the answers to many questions are covered there.

Reviewer: I'd call out:
- https://github.com/flowforge/flowforge/discussions
- Node-RED Forums

Let's go where the community is. Slack could be less prominent IMO

Author: I've added those in, I've left Slack at the bottom of the list though.
